------------------------------------------------------------
revno: 14172
revision-id: squid3@treenet.co.nz-20170621195439-l63xfsad58ghhhfu
parent: squidadm@squid-cache.org-20170615001633-wgrl5w8isv15o7gg
fixes bug: http://bugs.squid-cache.org/show_bug.cgi?id=4671
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 3.5
timestamp: Thu 2017-06-22 07:54:39 +1200
message:
  Bug 4671 pt2: GCC 7: raise FTP Gateway CTRL channel buffer to 16KB
  
  Fixes
   error: %s directive output may be truncated writing up to 8191 bytes
          into a region of size 1019
   note: snprintf output between 8 and 8199 bytes into a destination of
         size 1024
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20170621195439-l63xfsad58ghhhfu
# target_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# testament_sha1: eeb32b45efe5504eebeaae89088d4a81d807807c
# timestamp: 2017-06-21 20:50:58 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# base_revision_id: squidadm@squid-cache.org-20170615001633-\
#   wgrl5w8isv15o7gg
# 
# Begin patch
=== modified file 'src/clients/FtpGateway.cc'
--- src/clients/FtpGateway.cc	2017-05-29 04:37:41 +0000
+++ src/clients/FtpGateway.cc	2017-06-21 19:54:39 +0000
@@ -192,7 +192,7 @@
 
 #define FTP_LOGIN_NOT_ESCAPED   0
 
-#define CTRL_BUFLEN 1024
+#define CTRL_BUFLEN 16*1024
 static char cbuf[CTRL_BUFLEN];
 
 /*

